src/navsat_transform.cpp

```diff
@@ -134,54 +129,57 @@ void NavSatTransform::run()
     datumCallback(request, response);
   }
 
-  auto odom_sub = node_->create_subscription<nav_msgs::msg::Odometry>(
+  odom_sub = this->create_subscription<nav_msgs::msg::Odometry>(
     "odometry/filtered", 10, std::bind(&NavSatTransform::odomCallback, this, _1));
-  auto gps_sub = node_->create_subscription<sensor_msgs::msg::NavSatFix>(
+
+  gps_sub = this->create_subscription<sensor_msgs::msg::NavSatFix>(
     "gps/fix", 10, std::bind(&NavSatTransform::gpsFixCallback, this, _1));
 
   if (!use_odometry_yaw_ && !use_manual_datum_) {
-    imu_sub = node_->create_subscription<sensor_msgs::msg::Imu>(
+    imu_sub = this->create_subscription<sensor_msgs::msg::Imu>(
       "imu", 10, std::bind(&NavSatTransform::imuCallback, this, _1));
   }
 
-  auto gps_odom_pub =
-    node_->create_publisher<nav_msgs::msg::Odometry>("odometry/gps", rclcpp::QoS(10));
+  gps_odom_pub =
+    this->create_publisher<nav_msgs::msg::Odometry>("odometry/gps", rclcpp::QoS(10));
 
   if (publish_gps_) {
     filtered_gps_pub =
-      node_->create_publisher<sensor_msgs::msg::NavSatFix>("gps/filtered", rclcpp::QoS(10));
+      this->create_publisher<sensor_msgs::msg::NavSatFix>("gps/filtered", rclcpp::QoS(10));
   }
 
   // Sleep for the parameterized amount of time, to give
   // other nodes time to start up (not always necessary)
   rclcpp::sleep_for(std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::duration<double>(delay)));
 
-  rclcpp::Rate rate(frequency);
-  while (rclcpp::ok()) {
-    rclcpp::spin_some(node_);
+  int interval = (1/frequency) * 1000;  
```

Comment:
you have frequency defaulted to `10`, so this would be... 100 hz? I think you mea to move `s` to `ms` but the variable name didnt make that clear, can you change the name or do the ms conversion in the chrono call? Or better yet, use `chrono::seconds(interval)`

Reply:
Yes, good catch. I'll keep the variable name, but I like the approach of  `chrono::seconds(interval)`
to make it more clear.